extension-manifest-v3/src/pages/settings/components/custom-filters.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import { html, store } from 'hybrids';
+import { detectFilterType } from '@cliqz/adblocker';
+
+import * as converter from '/utils/dnr-converter.js';
+import CustomFiltersInput from '../store/custom-filters-input.js';
+import { asyncAction } from './devtools.js';
+
+function parseFilters(text = '') {
+  return text
+    .split('\n')
+    .map((f) => f.trim())
+    .filter(Boolean)
+    .reduce(
+      (filters, filter) => {
+        const filterType = detectFilterType(filter);
+        if (filterType === 1) {
+          // NETWORK
+          filters.networkFilters.add(filter);
+        } else if (filterType === 2) {
+          // COSMETIC
+          filters.cosmeticFilters.add(filter);
+        } else {
+          filters.errors.push(`Filter not supported: '${filter}'`);
+        }
+        return filters;
+      },
+      {
+        networkFilters: new Set(),
+        cosmeticFilters: new Set(),
+        errors: [],
+      },
+    );
+}
+
+async function submitFilters(host) {
+  const { networkFilters, cosmeticFilters } = host.filters;
+
+  // Update DNR
+  if (__PLATFORM__ !== 'firefox') {
+    const dnrRules = [];
+    const dnrErrors = [];
+    const results = await Promise.allSettled(
+      [...networkFilters].map((filter) => converter.convert(filter)),
+    );
+
+    for (const result of results) {
+      dnrErrors.push(...result.value.errors);
+      dnrRules.push(...result.value.rules);
+    }
+
+    // DNR validation is not required for Firefox, but it wont harm
+    if (dnrErrors.length) {
+      host.dnrErrors = dnrErrors;
+      return;
+    }
+
+    await chrome.runtime.sendMessage({
+      action: 'custom-filters:dnr',
+      dnrRules,
+    });
+
+    host.dnrRules = dnrRules;
+  }
+
+  // Update engine
+  await chrome.runtime.sendMessage({
+    action: 'custom-filters:engine',
+    filters: [
+      ...(__PLATFORM__ === 'firefox' ? networkFilters : []),
```

Comment:
we have to pass network filters also for DNR platforms are there are certain network filter options that affect cosmetic filters `$elemhide, $ehide, $specifichide, $shide, $generichide, $ghide`